sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -556,11 +579,13 @@ def refresh(self, **kwargs) -> None:
                         if need_refresh:
                             self._dict = configuration_settings_processed
                     if self._feature_flag_refresh_enabled:
-                        need_ff_refresh, self._refresh_on_feature_flags, feature_flags, filters_used = (
+                        need_ff_refresh, refresh_on_feature_flags, feature_flags, filters_used = (
                             client.refresh_feature_flags(
                                 self._refresh_on_feature_flags, self._feature_flag_selectors, headers, **kwargs
                             )
                         )
+                        if refresh_on_feature_flags:
+                            self._refresh_on_feature_flags = refresh_on_feature_flags
```

Comment:
Why is this change needed?

Reply:
It's been a while since I wrote this, it sort of looks like a possible bug fix? This verifies that refresh_on_feature_flags is never set to `None`. Thoughts?